modules/cache-common/src/test/java/org/opensearch/cache/common/tier/MockDiskCache.java

```diff
@@ -156,4 +163,48 @@ public Builder<K, V> setValueSerializer(Serializer<V, byte[]> valueSerializer) {
         }
 
     }
+
+    /**
+     * Provides a iterator over keys.
+     * @param <K> Type of key
+     * @param <V> Type of value
+     */
+    class CacheKeyIterator<K, V> implements Iterator<K> {
+        private final Iterator<Map.Entry<K, V>> entryIterator;
+        private final Map<K, V> cache;
+        private final RemovalListener<K, V> removalListener;
+        private K lastKey;
+
+        public CacheKeyIterator(Map<K, V> cache, RemovalListener<K, V> removalListener) {
+            this.entryIterator = cache.entrySet().iterator();
+            this.removalListener = removalListener;
+            this.cache = cache;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return entryIterator.hasNext();
+        }
+
+        @Override
+        public K next() {
+            if (!hasNext()) {
+                throw new NoSuchElementException();
+            }
+            Map.Entry<K, V> entry = entryIterator.next();
+            lastKey = entry.getKey();
+            return lastKey;
+        }
+
+        @Override
+        public void remove() {
+            if (lastKey == null) {
```

Comment:
Should this be swapped to "nextKey" or similar? It seems like this is referencing the next element in the cache's key. If you were to remove this, I think you be pruning all elements beyond the current leaf based on my reading not creating a sub tree including the current node. That being the case, this reads a bit confusing if we call it last key. 

Perhaps I am misreading?

Reply:
Yeah naming is confusing, I can call rename it to nextKey.